ctc_decoder/beam_search.py

```diff
@@ -42,8 +45,11 @@ def apply_lm(parent_beam, child_beam, labels, lm):
         c1 = labels[parent_beam.labeling[-1] if parent_beam.labeling else labels.index(' ')]  # first char
         c2 = labels[child_beam.labeling[-1]]  # second char
         lm_factor = 0.01  # influence of language model
-        bigram_prob = lm.get_char_bigram(c1, c2) ** lm_factor
-        child_beam.pr_text = parent_beam.pr_text * bigram_prob  # probability of char sequence
+        bigram_prob = lm_factor * np.log(lm.get_char_bigram(c1, c2))
+        if parent_beam.is_empty():
+            child_beam.pr_text = bigram_prob  # first char in beam
```

Comment:
no special case needed (no if else), as an empty beam is initialized with pr_text=log(1)=0, so adding parent_beam.pr_text + bigram_prob is OK

Reply:
Thanks for picking this up.